src/bit-systems/link-hover-menu.ts

```diff
@@ -0,0 +1,51 @@
+import { defineQuery, enterQuery, entityExists } from "bitecs";
+import type { HubsWorld } from "../app";
+import { Interacted, Link, LinkHoverMenu, LinkHoverMenuItem, HoveredRemoteRight } from "../bit-components";
+import { anyEntityWith } from "../utils/bit-utils";
+
+const NULL_EID = 0;
+
+const hoveredQuery = defineQuery([HoveredRemoteRight, Link]);
+const hoveredEnterQuery = enterQuery(hoveredQuery);
+const hoveredMenuItemQuery = defineQuery([HoveredRemoteRight, LinkHoverMenuItem]);
+const clickedMenuItemQuery = defineQuery([Interacted, LinkHoverMenuItem]);
+
+export function linkHoverMenuSystem(world: HubsWorld) {
+  // Assumes always only single LinkHoverMenu entity exists.
```

Comment:
For VR we are probably going to need to account for more than one, though maybe we can get away with only allowing a single cursor and just moving which hand it comes from.

Reply:
I will add TODO comment for now